homeassistant/components/iqvia/__init__.py

```diff
@@ -116,53 +104,58 @@ def __init__(self, client, sensor_types):
 
     async def async_update(self):
         """Update IQVIA data."""
-        from pyiqvia.errors import InvalidZipError
+        condition_method_mapping = {
+            (TYPE_ALLERGY_FORECAST,): [
+                (self._client.allergens.extended, TYPE_ALLERGY_FORECAST),
+                (self._client.allergens.outlook, TYPE_ALLERGY_OUTLOOK)
+            ],
+            (TYPE_ALLERGY_HISTORIC,): [(
+                self._client.allergens.historic, TYPE_ALLERGY_HISTORIC)],
+            (
+                TYPE_ALLERGY_TODAY,
+                TYPE_ALLERGY_TOMORROW,
+                TYPE_ALLERGY_YESTERDAY
+            ): [(self._client.allergens.current, TYPE_ALLERGY_INDEX)],
+            (TYPE_ASTHMA_FORECAST,): [(
+                self._client.asthma.extended, TYPE_ASTHMA_FORECAST)],
+            (TYPE_ASTHMA_HISTORIC,): [(
+                self._client.asthma.historic, TYPE_ASTHMA_HISTORIC)],
+            (
+                TYPE_ASTHMA_TODAY,
+                TYPE_ASTHMA_TOMORROW,
+                TYPE_ASTHMA_YESTERDAY
+            ): [(self._client.asthma.current, TYPE_ASTHMA_INDEX)],
+            (TYPE_DISEASE_FORECAST,): [(
+                self._client.disease.extended, TYPE_DISEASE_FORECAST)],
+        }
+
+        tasks = {}
+        for conditions, actions in condition_method_mapping.items():
+            if not any(True for c in conditions if c in self.sensor_types):
+                continue
+
+            for action, key in actions:
+                tasks[key] = action()
```

Comment:
Is it intended to only run the last action for each key?

Reply:
Don't we need to use `self._get_data` anymore?

Reply:
Negative: this should run for every action/key pair. This really only matters in the case of `TYPE_ALLERGY_FORECAST`: if the user provides that key, two coroutines need to be called (`self._client.allergens.extended` and `self._client.allergens.outlook`). In this case, their results will get stored for `TYPE_ALLERGY_FORECAST` and `TYPE_ALLERGY_OUTLOOK`, respectively.

`self._get_data` can be removed.

Reply:
As long as each task key has a single action it will work.